File: backend/src/models/Food.js
const mongoose = require('mongoose');

const foodSchema = new mongoose.Schema({
  loungeId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Lounge',
    required: [true, 'Lounge is required']
  },
  name: {
    type: String,
    required: [true, 'Food name is required'],
    trim: true
  },
  description: {
    type: String,
    trim: true
  },
  category: {
    type: String,
    enum: ['breakfast', 'lunch', 'dinner', 'snacks', 'drinks', 'dessert'],
    required: [true, 'Category is required']
  },
  price: {
    type: Number,
    required: [true, 'Price is required'],
    min: 0
  },
  image: {
    type: String
  },
  estimatedTime: {
    type: Number,
    required: [true, 'Estimated preparation time is required'],
    min: 1,
    comment: 'Time in minutes'
  },
  isAvailable: {
    type: Boolean,
    default: true
  },
  ingredients: [{
    type: String
  }],
  allergens: [{
    type: String
  }],
  isVegetarian: {
    type: Boolean,
    default: false
  },
  spicyLevel: {
    type: String,
    enum: ['none', 'mild', 'medium', 'hot', 'very-hot'],
    default: 'none'
  },
  rating: {
    average: {
      type: Number,
      default: 0,
      min: 0,
      max: 5
    },
    count: {
      type: Number,
      default: 0
    }
  }
}, {
  timestamps: true
});

// Index for efficient queries
foodSchema.index({ loungeId: 1, isAvailable: 1 });
foodSchema.index({ category: 1, isAvailable: 1 });

module.exports = mongoose.model('Food', foodSchema);

File: frontend/lib/models/food.dart
import 'package:equatable/equatable.dart';
import 'package:json_annotation/json_annotation.dart';
import 'lounge.dart';

part 'food.g.dart';

@JsonSerializable()
class Food extends Equatable {
  final String id;
  final String loungeId;
  final String name;
  final String? description;
  final String category;
  final double price;
  final String? image;
  final int estimatedTime;
  final bool isAvailable;
  final List<String>? ingredients;
  final List<String>? allergens;
  final bool isVegetarian;
  final String spicyLevel;
  final Rating rating;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Food({
    required this.id,
    required this.loungeId,
    required this.name,
    this.description,
    required this.category,
    required this.price,
    this.image,
    required this.estimatedTime,
    required this.isAvailable,
    this.ingredients,
    this.allergens,
    required this.isVegetarian,
    required this.spicyLevel,
    required this.rating,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Food.fromJson(Map<String, dynamic> json) => _$FoodFromJson(json);
  Map<String, dynamic> toJson() => _$FoodToJson(this);

  @override
  List<Object?> get props => [
        id,
        loungeId,
        name,
        description,
        category,
        price,
        image,
        estimatedTime,
        isAvailable,
        ingredients,
        allergens,
        isVegetarian,
        spicyLevel,
        rating,
        createdAt,
        updatedAt,
      ];
}

// Categories
enum FoodCategory {
  breakfast,
  lunch,
  dinner,
  snacks,
  drinks,
  dessert,
}

// Spicy Levels
enum SpicyLevel {
  none,
  mild,
  medium,
  hot,
  veryHot,
}
